source/frontend/src/features/properties/map/project/models.ts

```diff
@@ -1,81 +1,90 @@
-import { Api_Product, Api_Project } from 'models/api/Project';
+import { Api_Product, Api_Project, defaultProject } from 'models/api/Project';
 import { NumberFieldValue } from 'typings/NumberFieldValue';
 import { toTypeCode } from 'utils/formUtils';
 
 export class ProductForm {
-  id?: number;
+  id: number | undefined;
   code: string = '';
   description: string = '';
-  startDate?: string;
+  startDate: string | null = null;
   costEstimate: string = '';
-  costEstimateDate?: string;
-  objective?: string;
-  scope?: string;
-  rowVersion?: number;
+  costEstimateDate: string | null = null;
+  objective: string | null = null;
+  scope: string | null = null;
+  rowVersion: number | null = null;
 
-  toApi(parentId?: number): Api_Product {
+  toApi(parentId?: number | null): Api_Product {
     return {
       id: this.id,
-      parentProject: parentId !== undefined ? { id: parentId } : undefined,
+      parentProject: !!parentId
+        ? {
+            ...defaultProject,
+            id: parentId,
+          }
+        : null,
       code: this.code,
       description: this.description,
       startDate: this.startDate,
-      costEstimate: this.costEstimate !== undefined ? Number(this.costEstimate) : undefined,
+      costEstimate: !!this.costEstimate ? Number(this.costEstimate) : null,
       costEstimateDate: this.costEstimateDate,
       objective: this.objective,
       scope: this.scope,
       rowVersion: this.rowVersion,
+      acquisitionFiles: [],
     };
   }
 
   static fromApi(model: Api_Product): ProductForm {
     const newForm = new ProductForm();
-    newForm.id = model.id;
+    newForm.id = model.id ?? undefined;
     newForm.code = model.code || '';
     newForm.description = model.description || '';
     newForm.startDate = model.startDate;
     newForm.costEstimate = model.costEstimate?.toString() || '';
     newForm.costEstimateDate = model.costEstimateDate;
     newForm.objective = model.objective;
     newForm.scope = model.scope;
-    newForm.rowVersion = model.rowVersion;
+    newForm.rowVersion = model.rowVersion ?? null;
 
     return newForm;
   }
 }
 
 export class ProjectForm {
-  id?: number;
-  projectName?: string;
-  projectNumber?: string;
-  projectStatusType?: string;
-  region?: NumberFieldValue;
-  summary?: string;
-  rowVersion?: number;
+  id: number | undefined;
+  projectName: string | undefined = undefined;
+  projectNumber: string | undefined = undefined;
+  projectStatusType: string | undefined = undefined;
+  region: NumberFieldValue | undefined = undefined;
+  summary: string | undefined = undefined;
+  rowVersion: number | undefined = undefined;
```

Comment:
Would it be better if the forms also use null instead of undefined?

Reply:
I had the same thought, my reasoning here was more to do with backwards compatibility. We have to change all of the API* fields to support nulls, the same is not true on the forms. We already frequently reformat the data from the API to be easier to consume by the forms, so this seemed like a suitable way to reduce risk.

Reply:
The other reason being that the backend requires nulls since it does not support undefined. The frontend has no such limitations. Forcing the datatype of form fields to be x | null because of limitations on the backend seems like an abstraction leak.